packages/js/src/ui/components/Notification/NotificationList.tsx

```diff
@@ -40,14 +44,52 @@ const EmptyNotificationList = () => {
 type NotificationListProps = {
   mountNotification?: NotificationMounter;
   options?: ListNotificationsArgs;
+  filter?: NotificationFilter;
 };
 /* This is also going to be exported as a separate component. Keep it pure. */
 export const NotificationList = (props: NotificationListProps) => {
   const [data, { initialLoading, setEl, end }] = useNotificationsInfiniteScroll({ options: props.options });
+  const [newNotificationsCount, setNewNotificationsCount] = createSignal(0);
+  const { t } = useLocalization();
+  const style = useStyle();
+  const filter = createMemo(() => props.filter || { read: false });
+
+  useWebSocketEvent({
+    event: 'notifications.notification_received',
+    eventHandler: async (data) => {
+      const notification = data.result;
+      if (
+        !isSameFilter(filter(), {
+          read: notification.isRead,
+          archived: notification.isArchived,
+          tags: notification.tags,
+        })
+      ) {
+        return;
+      }
+
+      setNewNotificationsCount((c) => c + 1);
+    },
+  });
 
   return (
     <Show when={!initialLoading()} fallback={<NotificationListSkeleton count={8} />}>
       <Show when={data().length > 0} fallback={<EmptyNotificationList />}>
+        <Show when={!!newNotificationsCount()}>
+          <div
+            class={style(
+              'notificationListNewNotificationsNoticeContainer',
+              'nt-h-0 nt-w-full nt-flex nt-justify-center nt-top-4 nt-z-10'
+            )}
+          >
+            <Button
+              appearanceKey="notificationListNewNotificationsNotice__button"
+              class="nt-sticky nt-self-center nt-rounded-full nt-mt-1 hover:nt-bg-primary-600 nt-animate-fade-down"
+            >
+              {t('notifications.newNotifications', { notificationCount: newNotificationsCount() })}
+            </Button>
+          </div>
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** do you think showing here the `x` button will be useful? it might be annoying a little to see that this will hang forever until I refresh

Reply:
I'm not sure. I've noticed that apps like twitter don't have a way to hide this and always show it.